src/components/CustomDisciplineMenuItem.tsx

```diff
@@ -36,21 +44,78 @@ const ErrorText = styled.Text`
   margin: 10px;
 `
 
+const DeleteContainer = styled.View`
+  width: ${widthOfTrashButton};
+`
+
+const DeleteButton = styled(RectButton)`
+  height: ${wp('21%')}px;
+  margin: 0 16px 8px -16px;
+  align-items: center;
+  flex: 1;
+  justify-content: center;
+  background-color: ${props => props.theme.colors.lunesFunctionalIncorrectDark};
+  width: ${widthOfTrashButton};
+`
+
 interface CustomDisciplineMenuItemPropsType {
   apiKey: string
   selectedId: string | null
   setSelectedId: (input: string) => void
   navigation: StackNavigationProp<RoutesParamsType, 'Home'>
+  refresh: () => void
 }
 
 const CustomDisciplineMenuItem = ({
   apiKey,
   selectedId,
   setSelectedId,
-  navigation
+  navigation,
+  refresh
 }: CustomDisciplineMenuItemPropsType): JSX.Element => {
   const { data, loading } = useLoadGroupInfo(apiKey)
 
+  const [isModalVisible, setIsModalVisible] = useState<boolean>(false)
+
+  const renderRightAction = (progress: Animated.AnimatedInterpolation): ReactElement => {
+    const trans = progress.interpolate({
+      inputRange: [0, 1],
+      outputRange: [50, 0]
+    })
+
+    const showConfirmationModal = (): void => {
+      close()
+      setIsModalVisible(true)
+    }
+
+    return (
```

Comment:
I think it's nicer to keep the templates and the functions together for better readability.

Reply:
I'm with you, but `Swipeable` expects a function `renderRightActions` which needs to return the template.

Reply:
sorry i overseen that its inside this fkt. my mistake. then i'm fine :)